index.html

```diff
@@ -79,4 +79,238 @@
+  var chooseCat = Math.floor(Math.random() * 2);
+  var ind = (currentIndex % 10) + 1;
+  var newImage = (chooseCat) ?
+    'http://lorempixel.com/output/cats-q-c-640-480-' + ind + '.jpg':
+    'http://lorempixel.com/output/technics-q-c-640-480-' + ind + '.jpg';
+  currentIndex++;
+  return newImage;
+}
+
+var InfiniteScrollExample = React.createClass({
+  componentWillMount: function() {
+    this.secondsToWait = 2;
+    this.itemsToAdd = 3;
+  },
+
+  _loadMoreItems: function() {
+    this.setState({ loading: true });
+    // fake an async. ajax call with setTimeout
+    window.setTimeout(function() {
+      // add data
+      var currentItems = this.state.items;
+      for (var i = 0; i < this.itemsToAdd; i++) {
+        currentItems.push(generateItem());
+      }
+      this.setState({
+        items: currentItems,
+        loading: false,
+      });
+    }.bind(this), this.secondsToWait * 1000);
+  },
+
+  /**
+   * @return {Object}
+   */
+  getInitialState: function() {
+    var initialItems = [
+      'http://lorempixel.com/output/cats-q-c-640-480-9.jpg',
+      'http://lorempixel.com/output/cats-q-c-640-480-10.jpg',
+      'http://lorempixel.com/output/technics-q-c-640-480-10.jpg',
+    ];
+    return {
+      items: initialItems,
+      loading: false,
+    };
+  },
+
+  /**
+   * @return {Object}
+   */
+  _renderItems: function() {
+    return this.state.items.map(function(imageUrl, index) {
+      return (
+        <img
+          src={imageUrl}
+          alt="CATS AND ROBOTS... "
+          key={index}
+          className="infinite-scroll-example__list-item" />
+      );
+    });
+  },
+
+  /**
+   * @return {Object}
+   */
+  _renderLoadingMessage: function() {
+    if (this.state.loading) {
+      return (
+        <p className="infinite-scroll-example__loading-message">
+          Loading...
+        </p>
+      );
+    }
+  },
+
+  _renderWaypoint: function() {
+    if (!this.state.loading) {
+      return (
+        <Waypoint
+            onEnter={this._loadMoreItems}
+            threshold={2.0}
+        />
+      );
+    }
+  },
+
+  /**
+   * @return {Object}
+   */
+  render: function() {
+    return (
+      <div className="infinite-scroll-example">
+        <p className="infinite-scroll-example__count">
+          Items Loaded: {this.state.items.length}
+        </p>
+        {this._renderLoadingMessage()}
+        <div className="infinite-scroll-example__scrollable-parent">
+          {this._renderItems()}
+          {this._renderWaypoint()}
+        </div>
+        <p className="infinite-scroll-example__arrow" />
+      </div>
+    );
+  }
+});
+  </script>
+
+  <script type="text/jsx">
+var ExampleDemoAndCode = React.createClass({
+  /**
+   * @param {String} words
+   * @return {String}
+   */
+  _convertToTitle: function(words) {
+    return words.replace(/-|_/g, ' ')
+    .replace(/\b\s?(\S)/g, function(match, group) { return match.toUpperCase(); } );
+  },
+
+  _renderExampleComponent: function() {
```

Comment:
This could extrapolate the name of the component from the 'exampleName' prop, and use 'call' to instantiate it or something.